src/app/components/add-order/add-order.component.ts

```diff
@@ -138,28 +178,49 @@ export class AddOrderComponent implements OnInit {
     }
   }
   onSubmit() {
-    debugger;
     if (this.addOrderForm.valid) {
-      const orderData: Order = {
-        ...this.addOrderForm.value,
-        productList: this.productList.value,
+      const orderDataa = {
```

Comment:
Find a better way to map values instead of this

Note: you can check the form builder used to make an array of products in the form.